src/classes/Dml.cls

```diff
@@ -0,0 +1,104 @@
+public with sharing class Dml {
+    public void  BulkDMLOperation() {
+        List<Contact> updatedList = new List<Contact>();
+        List<Contact> conList = [Select Department , Description from Contact LIMIT 5];
+        for(Contact con : conList) {
+            if (con.Department == 'Finance') {
+                con.Description = 'New description';
+                // Add updated contact sObject to the list.
+                updatedList.add(con);
+            }
+        }
+        update updatedList;
+    }
+
+    public void AddingAndRetrieving(){
+        // Query existing account.
+        Account AcDetails= [SELECT Name,Industry 
+                            FROM Account WHERE Name='Salesforece' LIMIT 1];
+        // Write the old values the debug log before updating them.
+        System.debug('Account Name before update: ' + AcDetails.Name); // Name is Account Example
+        System.debug('Account Industry before update: ' + AcDetails.Industry);// Industry is not set
+        // Modify the two fields on the sObject.
+        AcDetails.Name = 'Account of the Day';
+        AcDetails.Industry = 'Technology';
+        // Persist the changes.
+        update AcDetails;
+        // Get a new copy of the account from the database with the two fields.
+        Account AcDetails1 = [SELECT Name,Industry FROM Account 
+                              WHERE Name='Account of the Day' LIMIT 1];
+        // Verify that updated field values were persisted.
+        System.assertEquals('Account of the Day', AcDetails1.Name);
+        System.assertEquals('Technology', AcDetails1.Industry);
+    }
+
+    //DML Inserting and Updating Records
+    public void InsertingAndUpdatingReceords(){
+        Account[] accts = new List<Account>();
+        for(Integer i=0;i<3;i++) {
+            Account a = new Account(Name='Acme' + i, BillingCity='San Francisco');
+            accts.add(a);
+        }
+        Account accountToUpdate;
+        try {
+            insert accts;            
+            // Update account Acme2.
+            accountToUpdate =[SELECT BillingCity FROM Account WHERE Name='Acme2' AND BillingCity='San Francisco'LIMIT 5];
+            // Update the billing city.
+            accountToUpdate.BillingCity = 'New York';
+            // Make the update call.
+            update accountToUpdate;
+        } catch(DmlException e) {
+            System.debug('An unexpected error has occurred: ' + e.getMessage());
+        }   
+        // Verify that the billing city was updated to New York.
+        Account afterUpdate =[SELECT BillingCity FROM Account WHERE Id=:accountToUpdate.Id];
+        System.assertEquals('New York', afterUpdate.BillingCity);  
+    }
+    //Creating Parent and Child Records in a Single Statement Using Foreign Keys
+    public void ParentAndChildRecords(){
```

Comment:
This one is for External ID not for the parent-child query. please fix the name of this method.
Methods for Parent-to-Child query and Child-to-Parent query are missing from this assignment.